framework/src/org/checkerframework/common/value/ValueVisitor.java

```diff
@@ -51,29 +79,60 @@ public Void visitAnnotation(AnnotationTree node, Void p) {
         }
 
         Element element = TreeInfo.symbol((JCTree) node.getAnnotationType());
-        if (!(element.toString().equals(ArrayLen.class.getName())
+
+        if (element.toString().equals(IntRange.class.getName())) {
+            if (args.size() == 2
+                    && args.get(0).getKind() == Kind.ASSIGNMENT
+                    && args.get(1).getKind() == Kind.ASSIGNMENT) {
+                ExpressionTree expFrom;
+                ExpressionTree expTo;
+
+                if (((AssignmentTree) args.get(0)).getVariable().toString().equals("from")) {
+                    expFrom = ((AssignmentTree) args.get(0)).getExpression();
+                    expTo = ((AssignmentTree) args.get(1)).getExpression();
+                } else {
+                    expFrom = ((AssignmentTree) args.get(1)).getExpression();
+                    expTo = ((AssignmentTree) args.get(0)).getExpression();
+                }
+
+                if (isIntLiteral(expFrom.getKind()) && isIntLiteral(expTo.getKind())) {
+                    try {
+                        long valueFrom = getIntLiteralValue(expFrom);
+                        long valueTo = getIntLiteralValue(expTo);
+                        if (valueFrom > valueTo) {
+                            checker.report(Result.failure("from.greater.than.to"), node);
+                            return null;
+                        }
+                    } catch (IllegalArgumentException e) {
```

Comment:
Can you write a comment about when this might occur?  Since you just checked that `expFrom` and `expTo` are int literals, it seems impossible that there would be a problem, in which case this catch clause can be omitted.

If you do retain the catch clause, then it should do something more drastic than just printing a stack trace, such as crashing the Checker Framework.